libbeat/common/mapstr.go

```diff
@@ -241,7 +241,7 @@ func (m MapStr) String() string {
 // An error is returned if underRoot is true and the value of ms.fields is not a
 // MapStr.
 func MergeFields(ms, fields MapStr, underRoot bool) error {
-	if ms == nil || fields == nil {
+	if ms == nil || fields == nil || len(fields) == 0 {
```

Comment:
This line can be just 

    if ms == nil || len(fields) == 0 {

because of how [`len`](https://golang.org/pkg/builtin/#len) works (it returns 0 when given a nil map).